definitions/npm/styled-components_v4.x.x/flow_v0.92.0-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,423 @@
+    getIds(): string[];
+    hasNameForId(id: string, name: string): boolean;
+    insertMarker(id: string): T;
+    insertRules(id: string, cssRules: string[], name: ?string): void;
+    removeRules(id: string): void;
+    css(): string;
+    toHTML(additionalAttrs: ?string): string;
+    toElement(): React$Element<*>;
+    clone(): Tag<T>;
+    sealed: boolean;
+  }
+
+  // The `any`/weak types in here all come from `styled-components` directly, since those definitions were just copied over
+  declare export class StyleSheet {
+    static get master() : StyleSheet;
+    static get instance() : StyleSheet;
+    static reset(forceServer? : boolean) : void;
+
+    id : number;
+    forceServer : boolean;
+    target : ?HTMLElement;
+    tagMap : {[string]: Tag<any>}; // eslint-disable-line flowtype/no-weak-types
+    deferred: { [string]: string[] | void };
+    rehydratedNames: { [string]: boolean };
+    ignoreRehydratedNames: { [string]: boolean };
+    tags: Tag<any>[]; // eslint-disable-line flowtype/no-weak-types
+    importRuleTag: Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    capacity: number;
+    clones: StyleSheet[];
+
+    constructor(?HTMLElement) : this;
+    rehydrate() : this;
+    clone() : StyleSheet;
+    sealAllTags() : void;
+    makeTag(tag : ?Tag<any>) : Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    getImportRuleTag() : Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    getTagForId(id : string): Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    hasId(id: string) : boolean;
+    hasNameForId(id: string, name: string) : boolean;
+    deferredInject(id : string, cssRules : string[]) : void;
+    inject(id: string, cssRules : string[], name? : string) : void;
+    remove(id : string) : void;
+    toHtml() : string;
+    toReactElements() : React$ElementType[];
+  }
+
+  declare export class KeyFrames {
+    id : string;
+    name : string;
+    rules : string[];
+
+    constructor(name : string, rules : string[]) : this;
+    inject(StyleSheet) : void;
+    toString() : string;
+    getName() : string;
+  }
+
+  // I think any is appropriate here?
+  // eslint-disable-next-line flowtype/no-weak-types
+  declare export var css : CSSConstructor;
+  declare export var keyframes : KeyFramesConstructor;
+  declare export var createGlobalStyle : CreateGlobalStyleConstructor
+  declare export var ThemeProvider: React$ComponentType<{children?: ?React$Node, theme: mixed | (mixed) => mixed}>
+
+  // This is a bit hard to read. Not sure how to make it more readable. I think adding line-breaks makes it worse.
+  declare type ThemeProps<T> = {
+    theme: T
+  }
+
+  declare export function withTheme<Theme, Config: {}, Instance>(Component: React$AbstractComponent<Config, Instance>): React$AbstractComponent<$Diff<Config, ThemeProps<Theme | void>>, Instance>
+
+  declare export type StyledComponent<Props, Theme, Instance> = React$AbstractComponent<Props, Instance> & InterpolatableComponent
```

Comment:
When testing this out in a real codebase, I'm seeing several instances of the following:

```
Cannot create CustomComponent element because InterpolatableComponent [1] is not a React component.
```

Is there a better way that we can handle this?

Reply:
@edahlseng Hmm. Can you provide the code that causes this error, if it's not too much code? Maybe we could make a test for that.

Reply:
I think a fix might be to do:

~~~js
declare class InterpolatableComponent extends React$Component<Props, any> {
  +styledComponentId: string;
}
~~~

I think using the interface might have been a _bigger_ hack than this anyway. I'll try this approach and see if the current tests pass, but it would be nice to add your case with the failing tests.

Reply:
Sounds good, I'll work on pulling out a minimal test case!

Reply:
So, this should probably be fixed by 121ec90. Would still be nice to have that test in there, so we know if it breaks in the future.